JhipsterXamarin/JhipsterXamarin/JhipsterXamarin/App.cs

```diff
@@ -1,20 +1,39 @@
-﻿using System.Net.Http;
+﻿using Akavache;
+using JhipsterXamarin.Models;
 using JhipsterXamarin.Services;
 using JhipsterXamarin.ViewModels;
 using MvvmCross;
 using MvvmCross.ViewModels;
+using System.Net.Http;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
 
 namespace JhipsterXamarin
 {
     public class App : MvxApplication
     {
         public override void Initialize()
         {
-            Mvx.IoCProvider.RegisterType<IAuthenticationService, AuthenticationService>();
+            Akavache.Registrations.Start("JhipsterXamarin");
+
+            var httpClient = new HttpClient();
+            var authenticationService = new AuthenticationService(httpClient);
+            Mvx.IoCProvider.RegisterSingleton(httpClient);
+            Mvx.IoCProvider.RegisterSingleton<IAuthenticationService>(authenticationService);
             Mvx.IoCProvider.RegisterType<IMyEntityService, MyEntityService>();
-            Mvx.IoCProvider.RegisterSingleton(new HttpClient());
 
-            RegisterAppStart<LoginViewModel>();
+            bool success = false;
+
+            try
+            {
+                var token = Task.Run(async () => await BlobCache.Secure.GetObject<JwtToken>("token")).Result;
+                success = Task.Run(async () => await authenticationService.SignIn(token)).Result;
+            }
+            catch { }
```

Comment:
it may be necessary to log the error (https://www.mvvmcross.com/documentation/fundamentals/logging)

Reply:
Created a static logger in e973fbf